lib/ansible/modules/web_infrastructure/ansible_tower/tower_credential.py

```diff
@@ -213,22 +218,57 @@ def main():
         tower_check_mode(module)
         credential = tower_cli.get_resource('credential')
         try:
-            params = module.params.copy()
+            params = {}
             params['create_on_missing'] = True
+            params['name'] = name
 
             if organization:
                 org_res = tower_cli.get_resource('organization')
                 org = org_res.get(name=organization)
                 params['organization'] = org['id']
 
-            if params['ssh_key_data']:
-                filename = params['ssh_key_data']
+            credential_type_res = tower_cli.get_resource('credential_type')
+            credential_type = credential_type_res.get(kind=module.params.get('kind'))
+            params['credential_type'] = credential_type['id']
+
+            if module.params.get('description'):
+                params['description'] = module.params.get('description')
+
+            if module.params.get('user'):
+                user_res = tower_cli.get_resource('user')
+                user = user_res.get(name=module.params.get('user'))
+                params['user'] = user['id']
+
+            if module.params.get('team'):
+                team_res = tower_cli.get_resource('team')
+                team = team_res.get(name=module.params.get('team'))
```

Comment:
If both team and organization is specified, then I would think we would filter the team by both, and remove the organization from params. This is for uniqueness (not sure if we've really dealt with that in other cases). Note: team is only valid on creation.